docs/source/python_spec.rst

```diff
@@ -96,7 +112,7 @@ C API which is called either when the refcount on the capsule named
       PyObject *type, *value, *traceback;
       PyErr_Fetch(&type, &value, &traceback);
 
-      DLManagedTensor *managed = (DLManagedTensor *)PyCapsule_GetPointer(self, "dltensor");
+      DLManagedTensorVersioned *managed = (DLManagedTensorVersioned *)PyCapsule_GetPointer(self, "dltensor");
```

Comment:
How can we distinguish between these two cases? For legacy capsules, we must cast to the older struct, right?

Reply:
> How can we distinguish between these two cases? For legacy capsules, we must cast to the older struct, right?

The offset of `dl_tensor.device` doesn't change with this ABI breaking change (and we also don't anticipate such a change in the future). So, casting it to either struct and accessing the field should work.

Reply:
Ahh, this is code created by the __producer__, not the consumer. When the consumer calls `obj.__dlpack__`, the producer creates a capsule, and the capsule's deleter function consists of the code here.